### PR TITLE
Add NeMo 2.0 section to the readme

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by jgerh** [comment]
Completed technical edit of NeMo 2.0 updates to ReadMe. LGTM.

**Review by ericharper — CHANGES_REQUESTED** [review]
Blocking temporarily, I think we should get feedback from @shashank3959 before merging.

File: README.md [hunk 1]
```diff
@@ -134,6 +140,29 @@ pre-trained model checkpoints.
...
+- **Modular Abstractions** - By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 simplifies adaptation and experimentation. This modular approach allows developers to more easily modify and experiment with different components of their models.
+
+- **Scalability** - NeMo 2.0 seamlessly scaling large-scale experiments across thousands of GPUs using [NeMo-Run](https://github.com/NVIDIA/NeMo-Run).
+
+- **Training Throughput** - NeMo 2.0 efficiently handles large-scale models and datasets. It manages large batch sizes and sequence lengths, distributing model components across GPUs through model parallelization.
+
+Overall, these enhancements make NeMo 2.0 a powerful, scalable, and user-friendly framework for AI model development.
+
+> [!IMPORTANT]  
+> NeMo 2.0 is currently supported by the LLM and VLM collections.
```
Comment (shashank3959) [inline README.md:158]: We have not defined VLMs so far. Good to dis-abbreviate.

File: README.md [hunk 2]
```diff
@@ -134,6 +140,29 @@ pre-trained model checkpoints.
...
+
+NVIDIA NeMo 2.0 introduces several significant improvements over its predecessor, NeMo 1.0, enhancing flexibility, performance, and scalability.
+
+- **Python-Based Configuration** - NeMo 2.0 transitions from YAML files to a Python-based configuration, providing more flexibility and control. This shift makes it easier to extend and customize configurations programmatically.
+
+- **Modular Abstractions** - By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 simplifies adaptation and experimentation. This modular approach allows developers to more easily modify and experiment with different components of their models.
+
+- **Scalability** - NeMo 2.0 seamlessly scaling large-scale experiments across thousands of GPUs using [NeMo-Run](https://github.com/NVIDIA/NeMo-Run).
+
+- **Training Throughput** - NeMo 2.0 efficiently handles large-scale models and datasets. It manages large batch sizes and sequence lengths, distributing model components across GPUs through model parallelization.
```
Comment (shashank3959) [inline README.md:153]: This isn't new with NeMo 2.0 right? Previous versions of NeMo could claim this capability too

File: README.md [hunk 3]
```diff
@@ -134,6 +140,29 @@ pre-trained model checkpoints.
...
+
+> [!IMPORTANT]  
+> NeMo 2.0 is currently supported by the LLM and VLM collections.
+
+### Get Started with NeMo 2.0
+
+- For more information about NeMo 2.0, see the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/index.html).
+- For an in-depth exploration of the main features of NeMo 2.0, see the [Feature Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/features/index.html#feature-guide).
+- To transition from NeMo 1.0 to 2.0, see the [Migration Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/migration/index) for step-by-step instructions.
+
```
Comment (shashank3959) [inline README.md:165]: In the Nemo Launcher section below, we should add a one liner clarification that NeMo-run is the recommended path forward for launching and managing experiments with NeMo 2.0

File: README.md [hunk 4]
```diff
@@ -134,6 +140,29 @@ pre-trained model checkpoints.
...
+## What's New in NeMo 2.0
+
+NVIDIA NeMo 2.0 introduces several significant improvements over its predecessor, NeMo 1.0, enhancing flexibility, performance, and scalability.
+
+- **Python-Based Configuration** - NeMo 2.0 transitions from YAML files to a Python-based configuration, providing more flexibility and control. This shift makes it easier to extend and customize configurations programmatically.
+
+- **Modular Abstractions** - By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 simplifies adaptation and experimentation. This modular approach allows developers to more easily modify and experiment with different components of their models.
+
+- **Scalability** - NeMo 2.0 seamlessly scaling large-scale experiments across thousands of GPUs using [NeMo-Run](https://github.com/NVIDIA/NeMo-Run).
+
```
Comment (shashank3959) [inline README.md:152]: Can we add that NeMo-run helps configure, execute and manage experiments.

File: README.md [hunk 5]
```diff
@@ -134,6 +140,29 @@ pre-trained model checkpoints.
...
+- **Training Throughput** - NeMo 2.0 efficiently handles large-scale models and datasets. It manages large batch sizes and sequence lengths, distributing model components across GPUs through model parallelization.
+
+Overall, these enhancements make NeMo 2.0 a powerful, scalable, and user-friendly framework for AI model development.
+
+> [!IMPORTANT]  
+> NeMo 2.0 is currently supported by the LLM and VLM collections.
+
+### Get Started with NeMo 2.0
+
+- For more information about NeMo 2.0, see the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/index.html).
```
Comment (shashank3959) [inline README.md:162]: Recommend linking directly to quick start assets as well: https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/quickstart.html

Comment (shashank3959) [reply README.md:162]: Infact, Quickstart should be the very first bullet. More information, migration etc. should be after that.

Comment (shashank3959) [reply README.md:162]: Second bullet could be the recipes folder we have: 
"[NeMo 2.0 Recipes](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/recipes) contains additional examples of launching large-scale runs using NeMo 2.0 and NeMo-Run."

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**